### PR TITLE
BB-139: throttle conductor depending on bucket queue

### PR DESCRIPTION
**Pull request by rachedbenmustapha**
- Fixes crash of lifecycle conductor when kafka is committing messages much slower than they're pulled from bucketd.
- Adds test that batches run in constant memory no matter the number of buckets (removing the circuit breaker causes the crash almost immediately).

### CODE REVIEW

File: .github/dockerfiles/syntheticbucketd/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM node:10
```
Comment (miniscruff) [inline .github/dockerfiles/syntheticbucketd/Dockerfile:1]: Should we use node 16 here

File: tests/utils/syntheticbucketd.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    if (marker) {
+        return Number.parseInt(marker.split('..|..')[1], 10) + 1;
+    }
+
+    return 0;
+}
+
+function generateAccountIds(accountsNumber) {
+    const accountPadding16 = '0000000000000000';
+    const accountPadding = accountPadding16 + accountPadding16 +
```
Comment (miniscruff) [inline tests/utils/syntheticbucketd.js]: You can use `.repeat(4);`

File: tests/utils/syntheticbucketd.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+function makeListing(marker, maxKeys) {
+    const accountIds = generateAccountIds(accountsNumber);
+    const index = getMarkerIndex(marker);
+    const ret = [];
+
+    const upperBound = Math.min(index + maxKeys, bucketNumber);
+
+    for (let i = index; i < upperBound; i += 1) {
+        const account = accountIds[i % accountsNumber];
+        const name = (`000000000000000000000000${i}`.slice(-24));
```
Comment (miniscruff) [inline tests/utils/syntheticbucketd.js]: can do `'0'.repeat(N)`

File: .github/dockerfiles/syntheticbucketd/Dockerfile (hunk 1 above)
Comment (rachedbenmustapha) [reply .github/dockerfiles/syntheticbucketd/Dockerfile:1]: It's a hotfix branch, we can't change the nodejs version.

Comment (miniscruff) [reply .github/dockerfiles/syntheticbucketd/Dockerfile:1]: Fair

**Review by miniscruff — APPROVED** [review]

**Review by tmacro — APPROVED** [review]

**Pull request merged**